allStructs/allStructs.ino

```diff
@@ -0,0 +1,156 @@
+    }yost_imu; // user defined data type
+
+
+unsigned long previousMillis = 0;
+const long interval = 500;
+
+  typedef struct { //initialize structure
+      float altitude;
+      float velocity; // velocity to be implemented later (a_1 - a_0) / time_inbetween_readings
+      float latitude;
+      float longitude; 
+      yost_imu rocket; 
+  } state;
+  
+void setup() {
+  Wire.begin();        // Join i2c bus
+  Serial.begin(9600);  // Start serial for output
+  myPressure.begin(); // Get sensor online
+  myGPS.begin(9600); // Get sensor online
+ yost.begin();
+ 
+  //Configure the sensor
+  myPressure.setModeAltimeter(); // Measure altitude above sea level in meters
+  //myPressure.setModeBarometer(); // Measure pressure in Pascals from 20 to 110 kPa
+
+  myPressure.setOversampleRate(7); // Set Oversample to the recommended 128
+  myPressure.enableEventFlags(); // Enable all three pressure and temp event flags 
+
+  typedef struct{ //initialize struct
+    float pitch;
+    float yaw;
+    float roll;
+    }euler_angles;
+
+    typedef struct{
+    float x;
+    float y;
+    float z;
+    }accel;
+
+    typedef struct{
+    euler_angles e_orient;
+    accel r_accel;
+    }yost_imu; // user defined data type
+
+    //yost_imu rocket = {{0,0,0},{0,0,0}}; // place holder vaules until sensor is updated
+   
+  typedef struct { //initialize structure
+      float altitude;
+      float velocity; // velocity to be implemented later (a_1 - a_0) / time_inbetween_readings
+      float latitude;
+      float longitude; 
+      yost_imu rocket;
+  } state;
+  state init_state = {myPressure.readAltitude(), 20.0, myGPS.latitude, myGPS.longitude,{{0,0,0},{0,0,0}} }; // random velocity picked
+  //state init_state = {myGPS.latitude, myGPS.longitude}; // random velocity picked
+
+  
+// function prototype
+void update_state(state *); // pointer to struct
+}
+
+void loop() {
+  state init_state;
+  update_state(&init_state); // function call pass memory address of init_state
+
+  
+  Serial.print("Altitude(m):");
+  Serial.print(init_state.altitude, 2);
+  Serial.println();
+  Serial.print("Latitude(Degrees):");
+  Serial.print(init_state.latitude, 2);
+  Serial.print("Longitude(Degrees):");
+  Serial.print(init_state.longitude, 2); //2?
+  Serial.println();
+ 
+
+  Serial.print("Pitch: ");
+  Serial.print(init_state.rocket.e_orient.pitch); 
+  Serial.println();
+
+  Serial.print("Yaw: ");
+  Serial.print(init_state.rocket.e_orient.yaw); 
+  Serial.println();
+
+  Serial.print("Roll: ");
+  Serial.print(init_state.rocket.e_orient.roll); 
+  Serial.println();
+
+  Serial.print(""); // \n
+  Serial.println();
+
+  Serial.print("x: ");
+  Serial.print(init_state.rocket.r_accel.x);
+  Serial.println();
+
+  Serial.print("y: ");
+  Serial.print(init_state.rocket.r_accel.y); 
+  Serial.println();
+
+  Serial.print("z: ");
+  Serial.print(init_state.rocket.r_accel.z); 
+  Serial.println();
+
+  Serial.print(""); // \n
+  Serial.println();
+  

```

Comment:
could you add a flag at the top like `PRINT_OUTPUT` or something like that, and check the flag before printing all of this to serial? just so that we don't print it during the actual flight.

Reply:
i added DISPLAY_OUTPUT as a function